internal/configuration/config.go

```diff
@@ -83,29 +92,38 @@ func newDefaultConfig() *Config {
 	config.Client.Redirects = 3
 	config.Client.Retries = 3
 	config.Statuspage.ApiRoot = "https://api.statuspage.io/v1"
+	config.Api.Port = 8080
 	return &config
 }
 
 // readEnvironmentVariables sets config values from the environment specifically only as described above
-func readEnvironmentVariables(config *Config) {
+func readEnvironmentVariables(config *Config) error {
 	apiKey, present := os.LookupEnv("REVERE_STATUSPAGE_APIKEY")
 	if present {
 		config.Statuspage.ApiKey = apiKey
 	}
+	stringPort, present := os.LookupEnv("REVERE_API_PORT")
```

Comment:
It does, but it does not actually proactively store values from the environment. `v.Get` works as expected, but `v.Unmarshall` to a struct has annoying behavior where the unmarshalled result won't include a value from the environment unless the value was also present in the config file. This means that the config file itself would be required to contain placeholders or something. 

I'm keeping Viper around because it does do some very nice file format conversions and things, though I wish it filled our requirements here